src/CarPlay/CPMessageListItem.cs

```diff
@@ -0,0 +1,50 @@
+//
+// CPMessageListItem.cs
+//
+// Authors:
+//	Whitney Schmidt <whschm@microsoft.com>
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under MIT License.
+//
+
+using System;
+using Foundation;
+using ObjCRuntime;
+
+#nullable enable
+
+namespace CarPlay {
+
+    public enum  CPMessageListItemType {
+        Identifier,
+        FullName,
+    }
+
+	public partial class CPMessageListItem {
+
+		public CPMessageListItem (string identifier, string text, CPMessageListItemLeadingConfiguration leadingConfiguration,
+
+            CPMessageListItemTrailingConfiguration? trailingConfiguration, string? detailText, string? trailingText) :
+                this (identifier, text, leadingConfiguration, trailingConfiguration, detailText, trailingText, CPMessageListItemType.Identifier)
+            {
+            }
+
+		public CPMessageListItem (string identifier, string text, CPMessageListItemLeadingConfiguration leadingConfiguration, CPMessageListItemTrailingConfiguration? trailingConfiguration,
+            string? detailText, string? trailingText, CPMessageListItemType type) : base (NSObjectFlag.Empty)
+        {
+            switch (type) {
+            case CPMessageListItemType.Identifier:
+                InitializeHandle (InitWithConversationIdentifier (identifier, text, leadingConfiguration, trailingConfiguration, detailText, trailingText),
+                    "initWithConversationIdentifier:text:leadingConfiguration:trailingConfiguration:detailText:trailingText:");
+                break;
+            case CPMessageListItemType.FullName:
+                InitializeHandle (InitWithFullName (identifier, text, leadingConfiguration, trailingConfiguration, detailText, trailingText),
+                    "initWithFullName:phoneOrEmailAddress:leadingConfiguration:trailingConfiguration:detailText:trailingText:");
+                break;
+            default:
+                throw new ArgumentException ("The 'CPMessageListItemType type' argument needs a value.");
```

Comment:
I think this is simpler and more accurate:

```suggestion
                throw new ArgumentOutOfRangeException (nameof (type));
```